driver/kubernetes/driver.go

```diff
@@ -218,9 +220,19 @@ func (d *Driver) Client(ctx context.Context) (*client.Client, error) {
 	opts = append(opts, client.WithContextDialer(func(context.Context, string) (net.Conn, error) {
 		return conn, nil
 	}))
+
 	if td, ok := exp.(client.TracerDelegate); ok {
 		opts = append(opts, client.WithTracerDelegate(td))
 	}
+
+	if d.Meta != nil {
+		opts = append(opts, client.WithGRPCDialOption(grpc.WithChainUnaryInterceptor(func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
```

Comment:
Why do we need this part?

Reply:
Wanted to be consistent for each driver and not just `docker`

Reply:
For other drivers this would be grpc headers so it is not the same. This is not configuring headers for the buildkit but only for the driver. And I don't think there is anything comparable in other drivers.

If anything, this is sort of like endpoint configuration.